test/unittests/t_util.cc

```diff
@@ -32,7 +38,94 @@ class ThreadDummy {
 };
 
 
-TEST(T_Util, ThreadProxy) {
+class T_Util : public ::testing::Test {
+ protected:
+  virtual void SetUp() {
+    const bool retval = MkdirDeep(string(sandbox), 0700);
+    ASSERT_TRUE(retval) << "failed to create string(sandbox)";
+  }
+
+  virtual void TearDown() {
+    const bool retval = RemoveTree(string(sandbox));
+    ASSERT_TRUE(retval) << "failed to remove string(sandbox)";
+  }
+
+ protected:
+  static void WriteBuffer(int fd) {
+    sleep(1);
+    write(fd, to_write, string(to_write).length());
+  }
+
+  static string CreateFileWithContent(const string &filename,
+      const string &content) {
+    string complete_path = string(sandbox) + "/" + filename;
+    FILE* myfile = fopen(complete_path.c_str(), "w");
+    fprintf(myfile, content.c_str());
+    fclose(myfile);
+    return complete_path;
+  }
+
+  static void LockFileTest(const string &filename, int *retval) {
+    *retval = LockFile(filename);
+  }
+
+  static string GetTimeString(time_t seconds, const bool utc) {
+    char buf[32];
+    struct tm ts;
+    if (utc) {
+      localtime_r(&seconds, &ts);
+    } else {
+      gmtime_r(&seconds, &ts);
+    }
+    strftime(buf, sizeof(buf), "%d %b %Y %H:%M:%S", &ts);
+    return string(buf);
+  }
+
+  static string GetRfcTimeString() {
+    time_t now = time(NULL);
+    char buf[32];
+    struct tm ts;
+    gmtime_r(&now, &ts);
+    strftime(buf, sizeof(buf), "%a, %d %b %Y %H:%M:%S %Z", &ts);
+    return string(buf);
+  }
+
+  static timeval CreateTimeval(int64_t tv_sec, int64_t tv_usec) {
+    timeval t;
+    t.tv_sec = tv_sec;
+    t.tv_usec = tv_usec;
+    return t;
+  }
+
+  static void SendSignal(int signal) {
+    BlockSignal(signal);
+    sleep(1);
```

Comment:
It's risky to rely on sleep.  Try to figure out if you find an order of events and threads that doesn't require sleep.